index.js

```diff
@@ -11,8 +9,50 @@ const randomstring = require('randomstring');
 const requestretry = require('requestretry');
 const tinytim = require('tinytim');
 const yaml = require('js-yaml');
+const _ = require('lodash');
+
+const MAXATTEMPTS = 5;
+const RETRYDELAY = 3000;
 const CPU_RESOURCE = 'beta.screwdriver.cd/cpu';
 const RAM_RESOURCE = 'beta.screwdriver.cd/ram';
+const TOLERATIONS_PATH = 'spec.tolerations';
+const AFFINITY_NODE_SELECTOR_PATH = 'spec.affinity.nodeAffinity.' +
+    'requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms';
+
+/**
+ * Parses nodeSelector config and update intended nodeSelector in tolerations
+ * and nodeAffinity.
+ * @method setNodeSelector
+ * @param {Object} podConfig k8s pod config
+ * @param {Object} nodeSelectors key-value pairs of node selectors
+ */
+function setNodeSelector(podConfig, nodeSelectors) {
+    if (!nodeSelectors || typeof nodeSelectors !== 'object') {
+        return;
+    }
+
+    const tolerations = _.get(podConfig, TOLERATIONS_PATH, []);
```

Comment:
Isn't this just `tolerations = []`
podConfig intially doesn't have `tolerations` : https://github.com/jithin1987/executor-k8s-vm/blob/430f085f25a8e9a39cd9ba13e11013a6c018cc91/index.js#L132-L146

Reply:
yes, it will, this is just to play safe in-case we introduce any must have tolerations from podConfig yaml template.